xbmc/addons/ContextMenuAddon.cpp

```diff
@@ -54,26 +54,26 @@ void CContextMenuAddon::ParseMenu(
 
   items.push_back(CContextMenuItem::CreateGroup(menuLabel, parent, menuId, addonInfo.ID()));
 
-  ELEMENTS subMenus;
-  if (CServiceBroker::GetAddonMgr().GetExtElements(elem, "menu", subMenus))
-    for (const auto& subMenu : subMenus)
-      ParseMenu(addonInfo, subMenu, menuId, anonGroupCount, items);
-
-  ELEMENTS elems;
-  if (CServiceBroker::GetAddonMgr().GetExtElements(elem, "item", elems))
+  for (unsigned int i = 0; i < elem->num_children; i++)
   {
-    for (const auto& elem : elems)
+    cp_cfg_element_t subElem = elem->children[i];
+    std::string elementName = subElem.name;
+    if (elementName.compare("menu") == 0)
```

Comment:
Isn't `elementName == "menu"` a lot easier to read?

Reply:
Indeed, but I was messing around with some other elements first. I will fix this.